create_per_gene_matrix.py

```diff
@@ -60,321 +177,319 @@ def get_worst_gene_csq_code_expr(vep_expr: hl.expr.StructExpression) -> hl.expr.
+    vp_mt = vp_mt.checkpoint(
+        f"gs://gnomad-tmp/compound_hets/chet_per_gene.filter_entries{'.test' if test else ''}.mt",
+        _read_if_exists=not overwrite,
+        overwrite=overwrite,
     )
-
-    vp_mt = vp_mt.annotate_cols(
-        pop=['all', vp_mt.pop]
-    )
-    vp_mt = vp_mt.explode_cols('pop')
-
-    vp_mt = vp_mt.explode_rows('vep')
-    vp_mt = vp_mt.transmute_rows(
-        **vp_mt.vep
+    
+    logger.info(
+        "Annotate variant pair MatrixTable with all relevant allele frequencies (max variant pair popmax AF <= AF cutoff "
+        "for the following cutoffs: %s) and the co-occurrence grouping of the variant pair "
+        "(chet cutoff: em.adj.p_chet >= %f , same hap cutoff: em.adj.p_chet <= %f)...",
+        ALLELE_FREQUENCY_CUTOFFS,
+        CHET_THRESHOLD,
+        SAME_HAP_THRESHOLD
     )
-
-    def get_grouped_phase_agg():
-        return hl.agg.group_by(
-            hl.case()
-                .when(~vp_mt.is_singleton_vp & (vp_mt.phase_info[vp_mt.pop].em.adj.p_chet > CHET_THRESHOLD), 1)
-                .when(~vp_mt.is_singleton_vp & (vp_mt.phase_info[vp_mt.pop].em.adj.p_chet < SAME_HAP_THRESHOLD), 2)
-                .default(3)
-            ,
-            hl.agg.min(vp_mt.csq)
+    vp_mt = vp_mt.annotate_rows(
+        af_cutoff=af_groups.filter(lambda af: vp_mt.popmax_af <= af),
+        chet_group=hl.case()
+        .when(
+            ~vp_mt.is_singleton_vp
+            & (vp_mt.phase_info["all"].em.adj.p_chet >= CHET_THRESHOLD),
+            "chet",
         )
-
-    vp_mt = vp_mt.group_rows_by(
-        'gene_id',
-        'gene_symbol'
-    ).aggregate(
-        all=hl.agg.filter(
-            vp_mt.x &
-            hl.if_else(
-                vp_mt.pop == 'all',
-                hl.is_defined(vp_mt.popmax_af) &
-                (vp_mt.popmax_af <= MAX_FREQ),
-                vp_mt.pop_af[vp_mt.pop] <= MAX_FREQ
-            ),
-            get_grouped_phase_agg()
-        ),
-        af_le_0_001=hl.agg.filter(
-            hl.if_else(
-                vp_mt.pop == 'all',
-                hl.is_defined(vp_mt.popmax_af) &
-                (vp_mt.popmax_af <= 0.001),
-                vp_mt.pop_af[vp_mt.pop] <= 0.001
-            )
-            & vp_mt.x,
-            get_grouped_phase_agg()
+        .when(
+            ~vp_mt.is_singleton_vp
+            & (vp_mt.phase_info["all"].em.adj.p_chet <= SAME_HAP_THRESHOLD),
+            "same_hap",
         )
+        .default("unphased"),
     )
 
-    vp_mt = vp_mt.checkpoint('gs://gnomad-tmp/compound_hets/chet_per_gene{}.2.mt'.format(
-        '.chr20' if chr20 else ''
-    ), overwrite=True)
+    logger.info(
+        "Exploding rows by VEP annotation (list of structs with 'gene_id', 'gene_symbol', 'csq' annotations), allele "
+        "frequency cutoffs..."
+    )
+    vp_mt = vp_mt.explode_rows("vep")
+    vp_mt = vp_mt.transmute_rows(**vp_mt.vep)
+    vp_mt = vp_mt.explode_rows(vp_mt.af_cutoff)
```

Comment:
I think we can actually modify this to be faster by moving this explode later and doing two aggregations see below